src/Microsoft.ML.FastTree/GamClassification.cs

```diff
@@ -158,49 +158,49 @@ protected override SchemaShape.Column[] GetOutputColumnsCore(SchemaShape inputSc
         }
     }
 
-    public class BinaryClassificationGamPredictor : GamPredictorBase, IPredictorProducing<float>
+    public class GamBinaryClassificationModelParameters : GamModelParametersBase, IPredictorProducing<float>
     {
-        public const string LoaderSignature = "BinaryClassGamPredictor";
+        internal const string LoaderSignature = "BinaryClassGamPredictor";
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
-        public BinaryClassificationGamPredictor(IHostEnvironment env, int inputLength, Dataset trainset,
+        public GamBinaryClassificationModelParameters(IHostEnvironment env, int inputLength, Dataset trainset,
```

Comment:
>GamBinaryClassificationModelParameters [](start = 15, length = 38)

Do you plan to add comments to all public constructors in separate PR?

Reply:
Separate PR sometimes means forever.

Reply:
I'm making the GAM constructors internal because they use `Dataset` object, which is in `Microsoft.ML.FastTree.Internal` that I don't think end user of API should be expected to know. I have filed issue #1948 to document it.

---
In reply to: [242421554](https://github.com/dotnet/machinelearning/pull/1901#discussion_r242421554) [](ancestors = 242421554)